src/language-js/print/type-annotation.js

```diff
@@ -556,7 +556,7 @@ function printTypeQuery({ node }, print) {
 function printTypePredicate(path, print) {
   const { node } = path;
   return [
-    node.asserts ? "asserts " : "",
+    node.asserts ? "asserts " : node.kind ? `${node.kind} ` : "",
```

Comment:
Flow has replaced the `asserts` boolean flag with a `kind` field that will either by `null`, or be `"asserts"` or `"implies"`.

Reply:
Did you mean `node.asserts` only exits in old Flow AST? Does it exists in TS?

Reply:
Never mind, TS use that as I checked.

Reply:
Maybe we can add `node.type` check here to make it clear.